rspace/src/test/scala/coop/rchain/rspace/history/SimplisticHistory.scala

```diff
@@ -306,7 +306,7 @@ final case class SimplisticHistory[F[_]: Sync](
     case (trie, path) => (trie, path.nodes)
   }
 
-  def reset(root: Blake2b256Hash): History[F] = this.copy(root = root)
+  def reset(root: Blake2b256Hash): F[History[F]] = Sync[F].delay(this.copy(root = root))
```

Comment:
In this case use of _pure_ here would be fine because only object instance is created with field assignment, but it makes sense to use _delay_ which seems more error prone for future changes.